flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/fusion/spec/RuntimeFilterFusionCodegenSpec.scala

```diff
@@ -0,0 +1,141 @@
+  private var buildType: RowType = _
+  private var probeType: RowType = _
+
+  private var buildComplete: String = _
+  private var filterTerm: String = _
+
+  override def setup(opFusionContext: OpFusionContext): Unit = {
+    super.setup(opFusionContext)
+    val inputContexts = fusionContext.getInputFusionContexts
+    assert(inputContexts.size == 2)
+    buildContext = inputContexts.get(0)
+    probeContext = inputContexts.get(1)
+
+    buildType = buildContext.getOutputType
+    probeType = probeContext.getOutputType
+  }
+
+  override def variablePrefix(): String = "rFilter"
+
+  override def doProcessProduce(codegenCtx: CodeGeneratorContext): Unit = {
+    // call build side first, then call probe side
+    buildContext.processProduce(codegenCtx)
+    probeContext.processProduce(codegenCtx)
+  }
+
+  override def doEndInputProduce(codegenCtx: CodeGeneratorContext): Unit = {
+    // call build side first, then call probe side
+    buildContext.endInputProduce(codegenCtx)
+    probeContext.endInputProduce(codegenCtx)
+  }
+
+  override def doProcessConsume(
+      inputId: Int,
+      inputVars: util.List[GeneratedExpression],
+      row: GeneratedExpression): String = {
+    if (inputId == buildInputId) {
+      buildComplete = newName("buildComplete")
+      opCodegenCtx.addReusableMember(s"private transient boolean $buildComplete;")
+      opCodegenCtx.addReusableOpenStatement(s"$buildComplete = false;")
+
+      filterTerm = newName("filter")
+      val filterClass = className[BloomFilter]
+      opCodegenCtx.addReusableMember(s"private transient $filterClass $filterTerm;")
+
+      s"""
+         |${className[Preconditions]}.checkState(!$buildComplete, "Should not build completed.");
+         |if ($filterTerm == null && !${row.resultTerm}.isNullAt(1)) {
+         |    $filterTerm = $filterClass.fromBytes(${row.resultTerm}.getBinary(1));
+         |}
+         |""".stripMargin
+    } else {
+      val Seq(probeKeyTerm, probeKeyWriterTerm) = newNames("probeKeyTerm", "probeKeyWriterTerm")
+      // project probe key row from input
+      val probeKeyExprs = probeIndices.map(idx => inputVars.get(idx))
+      val keyProjectionCode = getExprCodeGenerator
+        .generateResultExpression(
+          probeKeyExprs,
+          RowTypeUtils.projectRowType(probeType, probeIndices),
+          classOf[BinaryRowData],
+          probeKeyTerm,
+          outRowWriter = Option(probeKeyWriterTerm))
+        .code
+
+      val found = newName("found")
+      val consumeCode = fusionContext.processConsume(null, row.resultTerm)
+      s"""
+         |${className[Preconditions]}.checkState($buildComplete, "Should build completed.");
+         |
+         |boolean $found = true;
+         |if ($filterTerm != null) {
+         |  // compute the hash code of probe key
+         |  $keyProjectionCode
+         |  final int hashCode = $probeKeyTerm.hashCode();
+         |  if (!$filterTerm.testHash(hashCode)) {
```

Comment:
how about change here as follow:
```
if ($filterTerm.testHash(hashCode)) {
    ${row.code}
    $consumeCode
}
```

Reply:
No, the generated consume code may be too long, we should avoid splicing the same code into this code snippet repeatedly!